questions/class-name.md

```diff
@@ -0,0 +1,20 @@
+### Why React uses className over class attribute?
+
+#### Answer
+
+`class` is a reserved keyword in JavaScript and JSX is an extension of JavaScript. That's the primary reason why React uses className instead of class.
```

Comment:
`class` is a reserved word as an identifier i.e. variable name, but not as an object property (since ES5). So JSX compiles to `{ class: 'string' }` which is completely valid (Preact & Hyperapp do this).

The main reasons are actually:

- React in the early days had a design philosophy to align props with actual JS properties (`className` is the property) instead of native HTML despite JSX looking like HTML.
- It supported IE8 which did not have ES5 compatibility so `{ class: 'example' }` is invalid syntax (I believe)
- Since ES2015+: it does not work with destructuring without assigning it as a different name because it's a reserved word

Reply:
I also read somwehere that the next version of React will allow `<p classs="foo">bar<p>`, so this question might eventually be out of date.